app/src/main/kotlin/com/adesso/movee/scene/tvshow/TvShowViewModel.kt

```diff
@@ -62,12 +58,8 @@ class TvShowViewModel @Inject constructor(
     }
 
     private fun fetchNowPlayingTvShows() {
-        bgScope.launch {
-            val nowPlayingTvShowsResult = fetchNowPlayingTvShowsUseCase.run(UseCase.None)
-
-            onUIThread {
-                nowPlayingTvShowsResult.either(::handleFailure, ::postNowPlayingTvShows)
-            }
+        fetchNowPlayingTvShowsUseCase.invoke(viewModelScope, UseCase.None) {
+            it.either(::handleFailure, ::postNowPlayingTvShows)
```

Comment:
postNowPlayingTvShows() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postNowPlayingTvShows()